foxglove/cmd/utils.go

```diff
@@ -228,3 +230,12 @@ func maybeConvertToRFC3339(timestamp string) (string, error) {
 	}
 	return parsed.Format(time.RFC3339), nil
 }
+
+func AuthIsApiKey() (bool, error) {
```

Comment:
Looks like we never return an error, so can remove that from the signature

Reply:
I think it would be a bit more direct if we had,

```
TokenIsAPIKey(token string) bool
```

and called that in your Run method with the appropriate param, before calling `executeInfo`. So, read in the config value, validate it, and then call executeInfo when you know the config is valid.

This will help to minimize the presence of functions that cause the process to exit (i.e `dief`), outside of the Run methods. In general it's good to do that because functions that can randomly cause the process to exit are difficult to test.